internal/broker/broker.go

```diff
@@ -399,16 +399,19 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 		return AuthDenied, "", err
 	}
 
-	// The data type is unknown, so we need to directly call its marshal function rather than relying
-	// on json.Marshal, which would reflect the struct fields and use what is implemented for the basic types.
-	encoded, err := data.MarshalJSON()
+	encoded, err := json.Marshal(iaResponse)
 	if err != nil {
 		return AuthDenied, "", fmt.Errorf("could not marshal data: %v", err)
 	}
-	return access, string(encoded), nil
+
+	data := string(encoded)
+	if data == "null" {
+		data = ""
+	}
```

Comment:
Can't we prevent this earlier? Like checking zero'd `encoded` or implementing a special encoder?

Reply:
Didier and I thought about it, but it's not worth implementing an `emptyMessage` type as this only happens on a single occasion, IIRC.  As this is a very simple check, I guess it's okay as it is.